### PR TITLE
Merge ApiCapability class into ApiAbility

### PR DESCRIPTION
**Pull request by AntonKhorev**
There was an attempt to separate *abilities* and *capabilities* in https://github.com/openstreetmap/openstreetmap-website/commit/91fc65a2e3ad47414344a6d5dc03cb5fb2a08ce1 that didn't quite work out and is no longer necessary. The idea was "let's check user roles in *abilities* and oauth permissions in *capabilities*". This is explained in the commit comment and in https://github.com/openstreetmap/openstreetmap-website/pull/2023#discussion_r230071141. But as you can see *capabilities* still check roles as well.

There's an acknowledgement in https://github.com/openstreetmap/openstreetmap-website/commit/91fc65a2e3ad47414344a6d5dc03cb5fb2a08ce1 that this requires working against CanCanCan to some extent. The plan was to introduce `granted_capabilities` method that checks only *capabilities* and then use it in error reporting. However that method got removed in https://github.com/openstreetmap/openstreetmap-website/commit/0888f43d7b5a5e23874ff1a291049b90481d1273 before anything got merged into master.

What we ended up is three groups of ability granting:
1. ApiAbility before `if user&.active?`
2. ApiAbility after `if user&.active?`
3. ApiCapability

**1** are granted to everyone. **2** is a subset of **3** without token scope checks. **2** was used for authorizations without tokens such as basic auth, but that was removed. If you have a token, `ApiAbility.new(nil)` is executed that skips **2**. `ApiAbility.new(current_user)` that could run **2** is only executed if you don't have a token, but in this case you don't have `current_user`, so **2** is not going to run either.
https://github.com/openstreetmap/openstreetmap-website/blob/e2904daf7764928d33a67055349d5660040e16dd/app/controllers/api_controller.rb#L67-L71

Without something like basic auth we don't need **2** and we can replace it with **3** from *capabilities* and then delete the *capabilities* class.

### CODE REVIEW

File: app/abilities/api_ability.rb [hunk 1]
```diff
@@ -3,12 +3,16 @@
 class ApiAbility
   include CanCan::Ability
 
-  def initialize(user)
+  def initialize(token)
     can :read, [:version, :capability, :permission, :map]
 
     if Settings.status != "database_offline"
+      user = (User.find(token.resource_owner_id) if token)
```
Comment (tomhughes) [inline app/abilities/api_ability.rb:10]: I don't think the outer parentheses are needed here? They would be if `user` already had a value and you wanted to make sure it was overwritten but that's not the case here.

File: app/controllers/api_controller.rb [hunk 2]
```diff
@@ -64,11 +64,7 @@ def authorize(errormessage = "Couldn't authenticate you")
 
   def current_ability
     # Use capabilities from the oauth token if it exists and is a valid access token
-    if doorkeeper_token&.accessible?
-      ApiAbility.new(nil).merge(ApiCapability.new(doorkeeper_token))
-    else
-      ApiAbility.new(current_user)
-    end
+    ApiAbility.new((doorkeeper_token if doorkeeper_token&.accessible?))
```
[the remark this answers is not in the document]
Comment (AntonKhorev) [reply app/controllers/api_controller.rb:67]: This is where you get `current_user`, it requires a token:
https://github.com/openstreetmap/openstreetmap-website/blob/e2904daf7764928d33a67055349d5660040e16dd/app/controllers/api_controller.rb#L98

File: app/abilities/api_ability.rb (hunk 1 above)
Comment (AntonKhorev) [reply app/abilities/api_ability.rb]: removed parentheses

File: app/controllers/api_controller.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (AntonKhorev) [reply app/controllers/api_controller.rb]: restored `if`

**Review by tomhughes — APPROVED** [review]
This looks good now, thanks.

**Pull request merged**